### ISSUE TITLE
Bulk API not being called on Array input.

### ISSUE THREAD

**Issue by thegiles** [hammoaj/node-red-contrib-cloudantplus#6]
Passing an array in `msg.payload` is not resulting in a bulk document insert. Instead a single document is inserted with properties '0', '1', '2', etc. The node is configured for the insert operation, and to only take the `msg.payload`. I have also tried passing the array as the `msg` and unchecked the payload only option with no luck.

**Comment by ibmrcruicks** [comment]
Object.assign in    handleMessage --
```
            var data  = Object.assign({}, node.payonly ? msg.payload : msg);
```
is destroying the array .

Preserve array/object  property  during assign:

```
            var root = node.payonly ? "payload" : "msg";
            if(Object.prototype.toString.call( msg[root] ) === '[object Array]'){
              var data  = Object.assign([], msg[root]);
            }
            else {
              var data  = Object.assign({}, msg[root]);
            }
```

**Comment by Stwissel** [comment]
Fixed in upcoming 0.4

**Comment by hammoaj** [comment]
Should be fixed in latest 2.0.1 version
